bin/development-mode

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env php
+<?php
+/**
+ * @link      http://github.com/zendframework/ZendSkeletonApplication for the canonical source repository
+ * @copyright Copyright (c) 2005-2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+const CONFIG_CACHE_BASE = 'module-config-cache';
+
+$help = <<< EOH
+Enable/Disable development mode.
+
+Usage:
+
+development-mode [-h|--help] disable|enable
+
+--help|-h                    Print this usage message.
+disable                      Disable development mode.
+enable                       Enable development mode
+                             (do not use in production).
+
+To enable development mode, the following file MUST exist:
+
+- config/development.config.php.dist; this file will be copied to
+  config/development.config.php
+
+Optionally:
+
+- config/autoload/development.local.php.dist; this file will be copied to
+  config/autoload/development.local.php
+
+When disabling development mode:
+
+- config/development.config.php will be removed if it exists
+- config/autoload/development.local.php will be removed if it exists
+
+Additionally, both when disabling and enabling development mode, the
+script will remove the file cache/module-config-cache.php (or the file
+specified by the combination of the module_listener_options.cache_dir
+and module_listener_options.config_cache_key options).
+EOH;
+
+// Called without arguments
+if ($argc < 2) {
+    fwrite(STDERR, 'No arguments provided.' . PHP_EOL . PHP_EOL);
+    fwrite(STDERR, $help . PHP_EOL);
+    exit(1);
+}
+
+// Requested help
+if (in_array($argv[1], ['-h', '--help'], true)) {
+    echo $help, PHP_EOL;
+    exit(0);
+}
+
+if (! in_array($argv[1], ['disable', 'enable'], true)) {
+    fwrite(STDERR, 'Unrecognized argument.' . PHP_EOL . PHP_EOL);
+    fwrite(STDERR, $help . PHP_EOL);
+    exit(1);
+}
+
+$action = $argv[1];
+$return = $action();
+exit((int) $return);
```

Comment:
@webimpress Maybe, maybe not. `exit()` expects an integer, however, and I'd rather code defensively; we may enable strict mode in the future, and this will ensure that works.